helix-view/src/editor.rs

```diff
@@ -1180,7 +1184,7 @@ impl Editor {
             theme_loader,
             last_theme: None,
             last_selection: None,
-            registers: Registers::default(),
+            registers: Registers::new(conf.clipboard_provider.clone()),
```

Comment:
Now that the clipboard provider is a stateless value we can take advantage of a feature of the `arc_swap` that we use to hold Config and provide the Registers type just that value from config. In `Registers` the `clipboard_provider` field can be a `Box<dyn DynAccess<ClipboardProvider>>` (using `arc_swap::access::DynAccess`) and here in `editor.rs` we can set create this value with

```rust
Box::new(arc_swap::access::Map::new(
    Arc::clone(&config),
    |config: &Config| &config.clipboard_provider,
))
```

this way `registers.rs` only needs to know about this one config value rather than the entire `Config` struct and we can remove the update of the clipboard provider from `refresh_config` in this module. And the `clipboard_provider` field on `Registers` can remain private.

In places where the clipboard provider is read in `register.rs` we would access the field with `&self.clipboard_provider.load()` and functions like `read_from_clipboard` would take a `&ClipboardProvider`